src/vivarium_census_prl_synth_pop/components/person.py

```diff
@@ -255,8 +299,13 @@ def _perform_non_reference_person_moves(
 
         pop.loc[movers, "household_id"] = household_id_values
         pop.loc[movers, "housing_type"] = "Standard"
-        pop.loc[movers, "address_id"] = household_id_values.map(non_gq_address_ids)
         pop.loc[movers, "relation_to_household_head"] = "Other nonrelative"
+        # Update address and enforce address_id-state/puma link
+        pop.loc[movers, "address_id"] = household_id_values.map(non_gq_address_ids)
+        pop = pop.reset_index().set_index("address_id")
```

Comment:
I feel like there must be a simpler and less scary way to do this -- does this work?

```
pop.loc[movers, ["state", "puma"]] = pop.loc[movers, "address_id"].map(address_mapping)
```

Reply:
Can you map to two columns like this?

Reply:
I don't know, I didn't test it -- but even mapping them one at a time would be preferable to the current method, IMO

Reply:
Zeb, I tried so many things here and couldn't get it to work. I'll poke around a bit more. What's scary, though? All the resetting of index?

Jim, yep! Though there are some "rules" you must follow.

Reply:
Scary: resetting of index, and not limiting what you change to the `movers`. Plus, it's a bit long/hard to follow.

Reply:
@zmbc The reason your proposal doesn't work is b/c it's trying to map different indexes (left side is simulant, right side is address_id). That's why I go through the process of setting address_id as the index, then mapping, then setting back to simulant index.

I believe you can map from column but that is somehow scarier to me (there's no specific `left_on` and `right_on` like in `merge`). And I guess I could use merge but that seems clunky

Reply:
Oh, duh

```
pop.loc[movers, "state"] = pop["address_id"].map(address_mapping["state"])
pop.loc[movers, "puma"] = pop["address_id"].map(address_mapping["puma"])
```